appservice/src/deploy/deployZip.ts

```diff
@@ -14,14 +14,19 @@ import { waitForDeploymentToComplete } from './waitForDeploymentToComplete';
 export async function deployZip(context: IDeployContext, site: ParsedSite, fsPath: string, aspPromise: Promise<AppServicePlan | undefined>, pathFileMap?: Map<string, string>): Promise<void> {
     const kuduClient = await createKuduClient(context, site);
 
-    await runWithZipStream(context, {
+    const response = await runWithZipStream(context, {
         fsPath, site, pathFileMap,
         callback: async zipStream => {
-            await kuduClient.pushDeployment.zipPushDeploy(() => zipStream, { isAsync: true, author: 'VS Code' });
+            return await kuduClient.pushDeployment.zipPushDeploy(() => zipStream, { isAsync: true, author: 'VS Code' });
         }
     });
+    let locationUrl: string | undefined;
+    if (response) {
```

Comment:
How badly might things go if the `response` or `locationUrl` don't come through?

Reply:
Shouldn't break anything. Both of these properties are optional-- the `scm-deployment-id` is just for partner telemetry and the location url is the preferred way to track deployment progress, but our old method will still work if location url isn't defined.

Reply:
But I'll wrap this in a try catch just in-case response._response is undefined for some reason so it doesn't throw an errors